_posts/en/newsletters/2019-02-19-newsletter.md

```diff
@@ -0,0 +1,252 @@
+code and documentation changes in popular Bitcoin infrastructure
+projects.
+
+## Action items
+
+None this week.
+
+## News
+
+- **Discussion about tagging outputs to enable restricted features on
+  spending:** The [BIP118][] SIGHASH_NOINPUT_UNSAFE (noinput) proposal
+  allows the person generating a signature that authorizes the spend of
+  one UTXO to optionally allow that signature to be reused ("replayed")
+  for spending other UTXOs sent to the same public key.  This enables
+  new features when used with something like payment channels that all
+  contain the same public keys (see the proposed [Eltoo][] layer for
+  LN), but it also makes possible *replay attacks* that can result in a
+  loss of money when users reuse addresses.  For example: Alice uses a
+  noinput signature when spending some money she previously received to
+  one of her addresses.  Later someone else pays that same address
+  intending to send Alice some more money.  Now anyone with a copy of
+  Alice's earlier signature from the block chain can send that money to
+  the same person Alice paid earlier by replaying Alice's earlier
+  signature.
+
+    One way to prevent such accidents is simply by appending "UNSAFE" to
+    the name of the feature, encouraging developers to learn about the
+    protocol's nuances before they implement it in their tools.
+    However, some developers have been looking for additional ways to
+    prevent problems.  In December, Johnson Lau [proposed][lau output
+    tagging] only allowing noinput to be used if the output being spent
+    had been specially tagged at its creation to allow the use of
+    noinput.  This would only allow the feature to be used when both the
+    spender and receiver agreed (as in the case of a payment channel),
+    preventing any miscommunications or misunderstandings from ending in
+    loss of funds.
+
+    Renewed discussion [last week][nick output tagging] and this week
+    saw analysis of the impact this would have on proposed layer two
+    protocols such as Eltoo and [Channel Factories][].  Although the
+    proposal obviously increases complexity, the general opinion seems
+    to be that it doesn't fundamentally increase the cost or reduce the
+    effectiveness of the described proposals, although it could make
+    them a bit less private (because you'd communicate that noinput was
+    an option even in cases where you don't end up using
+    noinput---likely giving away that a second-layer protocol was in
+    use).
+
+- **Bitcoin Core preliminary hardware wallet support:** after months of
+  incremental improvements, this week saw the merge of the final set of
+  PRs needed for Bitcoin Core's master development branch to support
+  receiving and sending transactions in conjunction with a hardware
+  wallet via the [Hardware Wallet Interaction][HWI] (HWI) tool.  HWI is
+  not yet distributed with Bitcoin Core (although it is part of the
```

Comment:
Suggestion: remove the parentheses and re-order sentence, eg:

HWI is part of the Bitcoin Core project, but is not yet distributed with the Bitcoin Core software and is currently only accessible from the command line. It provides a solid...